### PR TITLE
Feature: login on Mia-Platform Console using password

### PR DESCRIPTION
**Pull request by danibix95**
#### Proposal

A new command is added to allow the cli to login on Mia-Platform Console. Using provided credentials it retrieves an access token that can be employed to call Console APIs. This token is saved in the configuration file.

Below is reported proposed command syntax:

```shell
miactl console login --username <username> --password <password> --app-id <app-id> --provider-id <provider-id>
```
where the arguments are:
- `username`: user identifier
- `password`: user secret
- `app-id`: identifier of which type of application is requesting to log on the Console
- `provider-id`: identifier of the authentication provider

#### Checklist

- [x] your branch will not cause merge conflict with `master`
- [x] run `go test`
- [x] tests are included
- [ ] the documentation is updated or intregrated accordingly with your changes
- [x] the code will follow the lint rules (`goimports`)
- [x] you are not committing extraneous files or sensible data

### CODE REVIEW

**Comment by davidebianchi** [comment]
As command, it is correct to set `miactl console login` instead of only `miactl login`?

Do we expect to make a login for every main resource?

File: cmd/console/login/login.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		ProviderID: providerID,
+	}
+	loginReq, err := JSONClient.NewRequest(http.MethodPost, "/oauth/token", data)
+	if err != nil {
+		return "", fmt.Errorf("error creating login request: %w", err)
+	}
+	var loginResponse tokenResponse
+
+	response, err := JSONClient.Do(loginReq, &loginResponse)
+	if err != nil || response.StatusCode != http.StatusOK {
```
Comment (davidebianchi) [inline cmd/console/login/login.go:94]: The JSONClient should check if status code is 2xx internally: https://github.com/davidebianchi/go-jsonclient/blob/master/main.go#L145

File: cmd/console/login/login_test.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+		cmd.Flags().Set("username", username)
+		cmd.Flags().Set("password", password)
+		cmd.Flags().Set("app-id", appID)
+		cmd.Flags().Set("provider-id", providerID)
+
+		require.Error(t, cmd.Execute(), "auth error:")
+
+		accessToken := viper.GetString("apitoken")
+		fmt.Println(accessToken)
+		require.Equal(t, expectedAccessToken, accessToken)
```
Comment (davidebianchi) [inline cmd/console/login/login_test.go:86]: Here it is not clear that `expectedAccessToken` is empty. Could we change the assertion to use empty, or inline set the expected as empty string?

**Review by davidebianchi — COMMENTED** [review]
As app-id, we should set it to `miactl`? Maybe as default? 
idk if it should be better to have a unique name identifying the cli

File: cmd/console/login/login.go (hunk 1 above)
Comment (danibix95) [reply cmd/console/login/login.go:94]: Thank you for pointing that out. Unfortunately, it was not specified in the library documentation that status codes were already handled by it.

File: cmd/console/login/login_test.go (hunk 2 above)
Comment (danibix95) [reply cmd/console/login/login_test.go:86]: Yes, you're right.

**Comment by danibix95** [comment]
> As command, it is correct to set `miactl console login` instead of only `miactl login`?
> 
> Do we expect to make a login for every main resource?

Actually, I introduced the intermediate `console` command to clarify that the login command is tied to the Console (not the resource). As you suggested, and taking into account that cli authentication is global (for all the resources), it is probably better to extract the login sub-command into a standalone one.

> As app-id, we should set it to miactl? Maybe as default?
> idk if it should be better to have a unique name identifying the cli

Considering the unique role of `miactl`, it may be more appropriate to set a unique name to identify it in the login process. Moreover, it would cut on the number of arguments required to login.

In conclusion, would be fine to use as interface the following?
```shell
miactl login --username <username> --password <password> --provider-id <provider-id>
```

File: cmd/console/login/login.go (hunk 1 above)
Comment (davidebianchi) [reply cmd/console/login/login.go]: Good point, I just added it. Thanks

File: cmd/login/login.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+			accessToken, err := login(baseURL, username, password, providerID)
+			if err != nil {
+				return err
+			}
+
+			// save current token for later commands
+			viper.Set("apitoken", accessToken)
+			fmt.Println("OK")
+			return viper.WriteConfig()
```
Comment (davidebianchi) [inline cmd/login/login.go:54]: If write config fails, user read OK and after the error stack trace.

Should we invert the command: before write config and later the ok message?

Comment (danibix95) [reply cmd/login/login.go]: Yes, that would be definitely better.

**Comment by danibix95** [comment]
I have re-based this branch onto `master` to resolve `go.sum` conflicts.

**Review by davidebianchi — APPROVED** [review]

**Comment by davidebianchi** [comment]
LGTM
